neotest-kotlin/app/src/test/kotlin/io/github/codymikol/neotestkotlin/framework/kotest/KotestTestRunnerFunctionalSpec.kt

```diff
@@ -0,0 +1,164 @@
+package io.github.codymikol.neotestkotlin.framework.kotest
+
+import io.github.codymikol.neotestkotlin.framework.TestRunResult
+import io.kotest.assertions.json.shouldEqualSpecifiedJson
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.matchers.types.shouldBeInstanceOf
+import kotlinx.serialization.json.Json
+
+class KotestTestRunnerFunctionalSpec :
+    FunSpec({
+        context("functional") {
+            test("run") {
+                val result = KotestTestRunner.run(listOf(KotestExampleSpec::class))
+                val actual = result.shouldBeInstanceOf<TestRunResult.Success>()
+                val actualJson = Json.encodeToString(actual.report)
+                actualJson shouldEqualSpecifiedJson
+                    """
+                    [
+                      {
+                        "name": "io.github.codymikol.neotestkotlin.framework.kotest.KotestExampleSpec",
+                        "nodes": [
+                          {
+                            "type": "test",
+                            "name": "pass",
+                            "status": {
+                              "status": "success"
+                            }
+                          },
+                          {
+                            "type": "test",
+                            "name": "fail",
+                            "status": {
+                              "status": "failure",
```

Comment:
The whole `stackTrace` is included here as well, but I omitted it for testing purposes. Going to add validation to ensure that the key exists at least